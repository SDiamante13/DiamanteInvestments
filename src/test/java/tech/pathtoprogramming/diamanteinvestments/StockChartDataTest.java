package tech.pathtoprogramming.diamanteinvestments;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class StockChartDataTest {

    @Test
    public void readLine() {
        StockChartData stockChartData = new StockChartData();

        String timeStamp = "1991-08-31";
        String open = "187.1500";
        String high = "188.4400";
        String low = "183.8850";
        String close = "185.6400";
        String volume = "82488674";
        stockChartData.readLine(timeStamp + "," + open + "," + high + "," + low + "," + close + "," + volume);

        assertThat(stockChartData.getDates().peek().getYear()).isEqualTo(1991);
    }
}