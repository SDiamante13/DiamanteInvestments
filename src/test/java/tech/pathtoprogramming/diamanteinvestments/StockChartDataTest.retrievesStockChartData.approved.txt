StockChartData{dates=[Sat Feb 02 00:00:00 EST 3924, Tue Jan 29 00:00:00 EST 3924, Mon Jan 28 00:00:00 EST 3924, Sun Jan 27 00:00:00 EST 3924, Sat Jan 26 00:00:00 EST 3924, Tue Jan 22 00:00:00 EST 3924, Mon Jan 21 00:00:00 EST 3924, Sun Jan 20 00:00:00 EST 3924, Sat Jan 19 00:00:00 EST 3924, Fri Jan 18 00:00:00 EST 3924, Tue Jan 15 00:00:00 EST 3924, Mon Jan 14 00:00:00 EST 3924, Sun Jan 13 00:00:00 EST 3924, Sat Jan 12 00:00:00 EST 3924, Fri Jan 11 00:00:00 EST 3924, Tue Jan 08 00:00:00 EST 3924, Mon Jan 07 00:00:00 EST 3924, Sun Jan 06 00:00:00 EST 3924, Sat Jan 05 00:00:00 EST 3924, Fri Jan 04 00:00:00 EST 3924, Tue Jan 01 00:00:00 EST 3924, Sun Dec 30 00:00:00 EST 3923, Sat Dec 29 00:00:00 EST 3923, Fri Dec 28 00:00:00 EST 3923, Thu Dec 27 00:00:00 EST 3923, Mon Dec 24 00:00:00 EST 3923, Sat Dec 22 00:00:00 EST 3923, Fri Dec 21 00:00:00 EST 3923, Thu Dec 20 00:00:00 EST 3923, Mon Dec 17 00:00:00 EST 3923, Sun Dec 16 00:00:00 EST 3923, Sat Dec 15 00:00:00 EST 3923, Fri Dec 14 00:00:00 EST 3923, Thu Dec 13 00:00:00 EST 3923, Mon Dec 10 00:00:00 EST 3923, Sun Dec 09 00:00:00 EST 3923, Sat Dec 08 00:00:00 EST 3923, Fri Dec 07 00:00:00 EST 3923, Thu Dec 06 00:00:00 EST 3923, Mon Dec 03 00:00:00 EST 3923, Sun Dec 02 00:00:00 EST 3923, Sat Dec 01 00:00:00 EST 3923, Sat Dec 01 00:00:00 EST 3923, Fri Nov 30 00:00:00 EST 3923, Tue Nov 27 00:00:00 EST 3923, Mon Nov 26 00:00:00 EST 3923, Sun Nov 25 00:00:00 EST 3923, Sat Nov 24 00:00:00 EST 3923, Fri Nov 23 00:00:00 EST 3923, Tue Nov 20 00:00:00 EST 3923, Mon Nov 19 00:00:00 EST 3923, Sun Nov 18 00:00:00 EST 3923, Sat Nov 17 00:00:00 EST 3923, Fri Nov 16 00:00:00 EST 3923, Tue Nov 13 00:00:00 EST 3923, Mon Nov 12 00:00:00 EST 3923, Sun Nov 11 00:00:00 EST 3923, Sat Nov 10 00:00:00 EST 3923, Fri Nov 09 00:00:00 EST 3923, Tue Nov 06 00:00:00 EST 3923, Mon Nov 05 00:00:00 EST 3923, Sun Nov 04 00:00:00 EDT 3923, Sat Nov 03 00:00:00 EDT 3923, Fri Nov 02 00:00:00 EDT 3923, Mon Oct 29 00:00:00 EDT 3923, Sun Oct 28 00:00:00 EDT 3923, Sat Oct 27 00:00:00 EDT 3923, Fri Oct 26 00:00:00 EDT 3923, Thu Oct 25 00:00:00 EDT 3923, Mon Oct 22 00:00:00 EDT 3923, Sun Oct 21 00:00:00 EDT 3923, Sat Oct 20 00:00:00 EDT 3923, Fri Oct 19 00:00:00 EDT 3923, Thu Oct 18 00:00:00 EDT 3923, Mon Oct 15 00:00:00 EDT 3923, Sun Oct 14 00:00:00 EDT 3923, Sat Oct 13 00:00:00 EDT 3923, Fri Oct 12 00:00:00 EDT 3923, Thu Oct 11 00:00:00 EDT 3923, Mon Oct 08 00:00:00 EDT 3923, Sun Oct 07 00:00:00 EDT 3923, Sat Oct 06 00:00:00 EDT 3923, Fri Oct 05 00:00:00 EDT 3923, Mon Oct 01 00:00:00 EDT 3923, Mon Oct 01 00:00:00 EDT 3923, Sun Sep 30 00:00:00 EDT 3923, Sat Sep 29 00:00:00 EDT 3923, Fri Sep 28 00:00:00 EDT 3923, Tue Sep 25 00:00:00 EDT 3923, Mon Sep 24 00:00:00 EDT 3923, Sun Sep 23 00:00:00 EDT 3923, Sat Sep 22 00:00:00 EDT 3923, Fri Sep 21 00:00:00 EDT 3923, Tue Sep 18 00:00:00 EDT 3923, Mon Sep 17 00:00:00 EDT 3923, Sun Sep 16 00:00:00 EDT 3923, Sat Sep 15 00:00:00 EDT 3923, Fri Sep 14 00:00:00 EDT 3923, Tue Sep 11 00:00:00 EDT 3923, Mon Sep 10 00:00:00 EDT 3923], opens=[187.15, 193.9, 194.14, 192.49, 193.61, 195.18, 196.1, 196.9, 196.16, 196.09, 197.53, 198.02, 195.09, 193.08, 193.11, 194.2, 193.63, 194.45, 190.21, 189.98, 190.33, 189.84, 190.9, 189.78, 189.92, 190.87, 191.49, 191.41, 189.89, 190.25, 189.57, 187.84, 187.7, 185.82, 183.97, 182.96, 182.35, 179.18, 176.38, 174.24, 175.52, 171.0, 169.35, 169.02, 166.91, 170.37, 171.88, 173.05, 170.91, 175.31, 176.04, 175.58, 176.65, 176.75, 181.42, 180.07, 178.2, 178.1, 176.81, 173.8, 173.79, 171.09, 172.25, 171.22, 172.02, 169.34, 172.62, 174.82, 174.2, 174.67, 174.55, 179.26, 177.52, 176.48, 176.48, 174.0, 176.51, 179.49, 180.07, 178.35, 175.18, 188.4, 188.28, 189.49, 187.84, 184.94, 179.69, 180.09, 177.38, 180.67, 178.52, 177.06, 175.07, 172.3, 177.14, 177.13, 178.88, 177.97, 177.32, 179.48], highs=[188.44, 194.4, 194.66, 193.5, 193.89, 195.41, 197.08, 197.68, 196.95, 196.63, 198.4, 199.62, 198.0, 194.72, 193.49, 195.99, 195.0, 194.76, 194.4, 190.05, 191.56, 190.32, 192.09, 191.08, 190.67, 190.9, 192.93, 191.52, 191.91, 190.38, 190.96, 189.5, 188.11, 186.03, 186.56, 184.12, 183.45, 182.44, 179.43, 176.82, 177.78, 174.23, 170.9, 171.17, 168.96, 171.38, 173.06, 173.67, 174.01, 175.42, 177.84, 177.57, 178.42, 179.07, 181.93, 182.34, 179.85, 179.72, 179.05, 177.99, 175.45, 174.21, 173.63, 174.3, 173.07, 172.03, 173.04, 175.2, 176.97, 177.08, 176.3, 179.69, 179.63, 179.38, 176.5, 176.1, 177.3, 180.13, 180.3, 180.24, 178.21, 188.85, 189.98, 189.92, 189.12, 187.85, 184.9, 180.59, 179.15, 181.1, 181.55, 177.68, 176.13, 175.1, 177.51, 178.54, 179.48, 179.69, 178.62, 180.75], lows=[183.88, 191.72, 193.17, 191.09, 192.83, 192.97, 193.5, 194.83, 195.89, 194.39, 197.0, 196.16, 194.85, 191.72, 191.42, 193.67, 193.59, 192.11, 190.18, 187.45, 189.23, 188.19, 188.97, 189.4, 188.9, 189.25, 190.82, 189.74, 189.88, 188.57, 188.65, 187.78, 186.3, 184.21, 183.53, 181.81, 181.59, 178.97, 176.21, 173.35, 175.46, 170.12, 167.9, 168.87, 166.83, 165.67, 170.65, 171.45, 169.93, 172.64, 175.19, 175.11, 174.8, 176.51, 178.14, 179.04, 177.6, 177.95, 175.8, 173.18, 172.68, 170.97, 170.82, 170.93, 170.34, 167.62, 169.05, 171.66, 174.15, 174.05, 173.86, 175.4, 177.13, 176.17, 173.82, 173.58, 173.98, 174.82, 177.34, 177.79, 173.54, 181.47, 187.61, 188.28, 187.48, 184.74, 179.5, 178.54, 175.82, 176.01, 178.32, 176.25, 173.74, 171.96, 173.48, 176.5, 177.05, 177.31, 176.55, 177.6], closes=[185.64, 192.53, 193.58, 193.15, 193.05, 193.6, 194.68, 194.83, 196.94, 195.89, 197.57, 198.11, 197.96, 194.71, 193.18, 195.71, 194.27, 192.32, 193.42, 189.43, 191.24, 189.95, 189.37, 190.4, 189.79, 189.97, 191.31, 190.64, 191.45, 189.69, 189.71, 188.01, 187.44, 184.8, 186.4, 182.41, 182.89, 181.82, 179.23, 176.65, 177.57, 173.97, 170.77, 170.29, 168.22, 166.89, 171.1, 173.44, 173.0, 172.88, 175.46, 175.84, 177.15, 178.72, 178.85, 180.71, 179.8, 178.39, 178.99, 177.49, 174.91, 173.66, 172.4, 173.75, 171.21, 170.69, 170.43, 171.96, 176.08, 174.79, 173.93, 175.49, 179.07, 177.97, 175.01, 175.74, 174.21, 176.3, 179.36, 178.18, 177.56, 182.91, 189.7, 189.46, 187.87, 187.65, 184.12, 180.19, 178.61, 176.38, 181.12, 177.23, 175.84, 174.49, 174.0, 176.57, 177.45, 179.46, 177.79, 177.97], volumes=[8248867.0, 4267214.0, 3404989.0, 4789980.0, 2891931.0, 3714957.0, 4648254.0, 5224281.0, 4023313.0, 5575186.0, 1.285384E7, 6683157.0, 7040418.0, 5269690.0, 6094369.0, 5340635.0, 4747765.0, 4089511.0, 6662839.0, 4338951.0, 4570482.0, 4879436.0, 4301422.0, 3841541.0, 4055260.0, 2404834.0, 3963001.0, 3813448.0, 4653861.0, 5094140.0, 5441291.0, 5379049.0, 6010837.0, 4362751.0, 6617792.0, 5376354.0, 4934028.0, 7052996.0, 6384131.0, 7982924.0, 7733475.0, 5693490.0, 4484601.0, 5113095.0, 5849912.0, 7062525.0, 5715696.0, 4381664.0, 5598010.0, 6424402.0, 5930286.0, 5476437.0, 5754935.0, 5251698.0, 5145608.0, 5674311.0, 4755109.0, 4369801.0, 4239077.0, 5726667.0, 4852791.0, 5302028.0, 4959461.0, 5216453.0, 5186108.0, 5629441.0, 6692180.0, 6458894.0, 4617274.0, 5672538.0, 6314911.0, 5843618.0, 5182694.0, 6725757.0, 1.0925946E7, 6089575.0, 8426792.0, 9037019.0, 5895305.0, 6560206.0, 1.124888E7, 8175581.0, 4528002.0, 4576650.0, 6079446.0, 6081388.0, 5300394.0, 4382069.0, 5144959.0, 5494579.0, 5272275.0, 4208424.0, 4631187.0, 6117215.0, 6606288.0, 4696485.0, 4362259.0, 4367562.0, 5203667.0, 5468685.0], timeSeries=TIME_SERIES_DAILY, interval=60min, date=Mon Sep 10 00:00:00 EDT 3923}